community/kernel/src/main/java/org/neo4j/kernel/impl/store/record/PropertyRecord.java

```diff
@@ -99,37 +103,78 @@ public long getRelId()
     public int size()
     {
         int result = 0;
-        for ( PropertyBlock blockRecord : blockRecords )
+        for ( int i = 0; i < blockRecordsCursor; i++ )
         {
-            result += blockRecord.getSize();
+            result += blockRecords[i].getSize();
         }
         return result;
     }
 
-    public List<PropertyBlock> getPropertyBlocks()
+    public int numberOfProperties()
     {
-        return blockRecords;
+        return blockRecordsCursor;
+    }
+
+    @Override
+    public Iterator<PropertyBlock> iterator()
+    {
+        blockRecordsIteratorCursor = 0;
+        return this;
+    }
+
+    @Override
+    public boolean hasNext()
+    {
+        return blockRecordsIteratorCursor < blockRecordsCursor;
+    }
+
+    @Override
+    public PropertyBlock next()
+    {
+        if ( !hasNext() )
+        {
+            throw new NoSuchElementException();
+        }
+        return blockRecords[blockRecordsIteratorCursor++];
+    }
+
+    @Override
+    public void remove()
+    {
+        if ( blockRecordsIteratorCursor == 0 )
```

Comment:
This `remove()` method comes from `Iterator` and so the item to be removed is the one we just iterator to in `next()`. `blockRecordsIteratorCursor` is incremented in `next()` and so we know that we can remove the "current iterator item" if we're not right at the beginning.

Although the current implementation breaks the `Iterator#contract` in that two consecutive calls to `remove()` might succeed and remove two items. There has to be a call to `next` in between. I'll fix that.